src/olympia/devhub/tests/test_tasks.py

```diff
@@ -539,6 +539,43 @@ def test_xpi_autoclose_is_enabled(self, subprocess_mock):
 
         assert '--disable-xpi-autoclose' not in self.FakePopen.get_args()
 
+    @override_switch('enable-mv3-submissions', active=False)
+    def test_mv3_submissions_waffle_disabled(self):
+        with mock.patch('olympia.devhub.tasks.subprocess') as subprocess_mock:
+            subprocess_mock.Popen = self.FakePopen
+
+            tasks.run_addons_linter(
+                path=self.valid_path, channel=amo.RELEASE_CHANNEL_LISTED
+            )
+
+            assert '--max-manifest-version=3' not in self.FakePopen.get_args()
+            assert '--max-manifest-version=2' in self.FakePopen.get_args()
+
+        mv3_path = get_addon_file('webextension_mv3.xpi')
+        result = tasks.run_addons_linter(mv3_path, channel=amo.RELEASE_CHANNEL_LISTED)
+        assert json.loads(result or '{}').get('errors') == 1
+
+    @override_switch('enable-mv3-submissions', active=True)
+    def test_mv3_submission_enabled(self):
+        with mock.patch('olympia.devhub.tasks.subprocess') as subprocess_mock:
+            subprocess_mock.Popen = self.FakePopen
+
+            tasks.run_addons_linter(
+                path=self.valid_path, channel=amo.RELEASE_CHANNEL_LISTED
+            )
+
+            assert '--max-manifest-version=3' in self.FakePopen.get_args()
```

Comment:
Should we also be asserting that `'--max-manifest-version=2' not in self.FakePopen.get_args()`, similar to what's being done in the test above?

Reply:
Not really sure testing for it in either test is necessary (I added the line when debugging) but I suppose it makes sense to do it in both tests consistently. https://github.com/mozilla/addons-server/pull/17471/commits/b15d08d6cb3b569da8f27609d09c2f43382783f3